src/util/GeoSparqlHelpers.h

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021, University of Freiburg,
+// Author: Hannah Bast <bast@cs.uni-freiburg.de>
+// Chair of Algorithms and Data Structures
+
+#ifndef QLEVER_GEOSPARQLHELPERS_H
+#define QLEVER_GEOSPARQLHELPERS_H
+
+#include <string>
+
+// TODO: How can I declare the lambdas in the .h file and implement them in the
+// .cpp file? The best I managed so far is define them with inline in the .f
+// file. If I define them without inline in the .cpp file, I get a multiple
+// definition error.
+//
+// namespace ad_utility {
+//
+// // Longitude of a WKT point (it's the first coordinate).
+// std::function<double(const std::string& p)> wktLongitude;
+//
+// // Latitude of a WKT point (it's the second coordinate).
+// std::function<double(const std::string& p)> wktLatitude;
+//
+// // Distance between two WKT points.
+// std::function<double(const std::string& p1, const std::string& p2)> wktDist;
+//
+// }  // namespace ad_utility
+
+// Helper function that checks whether a string looks like a WKT pointC (just
+// checks the beginning and that it has the minimal length).
+namespace {
+bool is_wkt_point(const std::string& p) {
+  return p.size() >= 12 && p[0] == '"' && (p[1] == 'P' || p[1] == 'p') &&
+         (p[2] == 'O' || p[2] == 'o') && (p[3] == 'I' || p[3] == 'i') &&
+         (p[4] == 'N' || p[4] == 'n') && (p[5] == 'T' || p[5] == 't') &&
+         p[6] == '(';
+}
+
+// Parse a single WKT Point of the form "POINT(<double> <double>)", where the
+// quote are included, the POINT must be uppercase, there is an arbitrary
+// amount of whitespace after the comma, and arbitrary contents after the
+// final quote. If the string does not match this format, a pair of
+// `quiet_NaN` is returned
+std::pair<double, double> parse_wkt_point(const std::string& p) {
+  auto error_result = std::pair(std::numeric_limits<double>::quiet_NaN(),
+                                std::numeric_limits<double>::quiet_NaN());
+  if (!is_wkt_point(p)) {
+    return error_result;
+  }
+  const char* beg = p.c_str() + 7;
+  char* end;
+  double lng = std::strtod(beg, &end);
+  if (end == nullptr) return error_result;
+  beg = end + 1;
+  double lat = std::strtod(beg, &end);
+  if (end == nullptr || *end != ')' || *(end + 1) != '"') return error_result;
```

Comment:
The `*end` should be always fine because you are getting a null-terminated string.
But `*(end + 1)` is an invalid read, e.g. if you have `"POINT(12.0 13.0`
It might be that this never happens because we always have at least the trailing quotation mark, but checking that end +1 is still valid is not too expensive (also in line 53/54 when you start the new round).

Reply:
I thought about this and I think it's OK. I added a comment that explains it.